real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/FullLoggingTest.java

```diff
@@ -0,0 +1,206 @@
+package com.roboclub.robobuggy.nodes.sensors;
+
+import com.roboclub.robobuggy.messages.BrakeStateMessage;
+import com.roboclub.robobuggy.messages.EncoderMeasurement;
+import com.roboclub.robobuggy.messages.GpsMeasurement;
+import com.roboclub.robobuggy.messages.GuiLoggingButtonMessage;
+import com.roboclub.robobuggy.messages.ImageMessage;
+import com.roboclub.robobuggy.messages.MagneticMeasurement;
+import com.roboclub.robobuggy.messages.SteeringMeasurement;
+import com.roboclub.robobuggy.ros.Message;
+import com.roboclub.robobuggy.ros.MessageListener;
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ros.Publisher;
+import com.roboclub.robobuggy.ros.Subscriber;
+import com.roboclub.robobuggy.simulation.SensorPlayer;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * A class that checks to see if we can log and play back correctly
+ */
+public class FullLoggingTest {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Each class should declare at least one constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2381518032/issues/source?bid=2938665&fileBranchId=3199438#l31)